### ISSUE TITLE
cuDNN error: CUDNN_STATUS_NOT_SUPPORTED on Colab

### ISSUE THREAD

**Issue by CathayPacific358** [open-mmlab/mmdetection#3519]
**Describe the bug**
When I try to implement a training with Libra_RCNN on Colab, this cuDNN error occurred, which is possibly a batch size problem. However, in the new version of mmdetection I haven't figure out any way to modify parameters like batch size or learning rate. The batch_size = num_gpus * samples_per_gpu for non-distributed training doesnt help. I wanna know exactly how to modify the batch size or learning rate for single GPU.

**Reproduction**
1. What command or script did you run?
```
python tools/train.py ./configs/libra_rcnn/libra_faster_rcnn_x101_64x4d_fpn_1x_voc.py
```
2. Did you make any modifications on the code or config? Did you understand what you have modified?
The modification only changing the data format from coco to voc, which is not relevant to the learning rate.
3. What dataset did you use?
Dataset in voc format is used.

**Environment**

1. Please run `python mmdet/utils/collect_env.py` to collect necessary environment information and paste it here.
sys.platform: linux
Python: 3.6.9 (default, Jul 17 2020, 12:50:27) [GCC 8.4.0]
CUDA available: True
CUDA_HOME: /usr/local/cuda
NVCC: Cuda compilation tools, release 10.1, V10.1.243
GPU 0: Tesla P100-PCIE-16GB
GCC: gcc (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0
PyTorch: 1.4.0
PyTorch compiling details: PyTorch built with:
  - GCC 7.3
  - Intel(R) Math Kernel Library Version 2019.0.4 Product Build 20190411 for Intel(R) 64 architecture applications
  - Intel(R) MKL-DNN v0.21.1 (Git Hash 7d2fd500bc78936d1d648ca713b901012f470dbc)
  - OpenMP 201511 (a.k.a. OpenMP 4.5)
  - NNPACK is enabled
  - CUDA Runtime 10.1
  - NVCC architecture flags: -gencode;arch=compute_37,code=sm_37;-gencode;arch=compute_50,code=sm_50;-gencode;arch=compute_60,code=sm_60;-gencode;arch=compute_61,code=sm_61;-gencode;arch=compute_70,code=sm_70;-gencode;arch=compute_75,code=sm_75;-gencode;arch=compute_37,code=compute_37
  - CuDNN 7.6.3
  - Magma 2.5.1
  - Build settings: BLAS=MKL, BUILD_NAMEDTENSOR=OFF, BUILD_TYPE=Release, CXX_FLAGS= -Wno-deprecated -fvisibility-inlines-hidden -fopenmp -DUSE_FBGEMM -DUSE_QNNPACK -DUSE_PYTORCH_QNNPACK -O2 -fPIC -Wno-narrowing -Wall -Wextra -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wno-sign-compare -Wno-unused-parameter -Wno-unused-variable -Wno-unused-function -Wno-unused-result -Wno-strict-overflow -Wno-strict-aliasing -Wno-error=deprecated-declarations -Wno-stringop-overflow -Wno-error=pedantic -Wno-error=redundant-decls -Wno-error=old-style-cast -fdiagnostics-color=always -faligned-new -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Wno-stringop-overflow, DISABLE_NUMA=1, PERF_WITH_AVX=1, PERF_WITH_AVX2=1, PERF_WITH_AVX512=1, USE_CUDA=ON, USE_EXCEPTION_PTR=1, USE_GFLAGS=OFF, USE_GLOG=OFF, USE_MKL=ON, USE_MKLDNN=ON, USE_MPI=OFF, USE_NCCL=ON, USE_NNPACK=ON, USE_OPENMP=ON, USE_STATIC_DISPATCH=OFF, 

TorchVision: 0.5.0
OpenCV: 4.1.2
MMCV: 0.6.2
MMDetection: 2.1.0+e7537a4
MMDetection Compiler: GCC 7.5
MMDetection CUDA Compiler: 10.1

**Error traceback**
If applicable, paste the error trackback here.
```
Traceback (most recent call last):
  File "tools/train.py", line 153, in <module>
    main()
  File "tools/train.py", line 149, in main
    meta=meta)
  File "/content/mmdetection/mmdet/apis/train.py", line 128, in train_detector
    runner.run(data_loaders, cfg.workflow, cfg.total_epochs)
  File "/usr/local/lib/python3.6/dist-packages/mmcv/runner/epoch_based_runner.py", line 122, in run
    epoch_runner(data_loaders[i], **kwargs)
  File "/usr/local/lib/python3.6/dist-packages/mmcv/runner/epoch_based_runner.py", line 43, in train
    self.call_hook('after_train_iter')
  File "/usr/local/lib/python3.6/dist-packages/mmcv/runner/base_runner.py", line 282, in call_hook
    getattr(hook, fn_name)(self)
  File "/usr/local/lib/python3.6/dist-packages/mmcv/runner/hooks/optimizer.py", line 21, in after_train_iter
    runner.outputs['loss'].backward()
  File "/usr/local/lib/python3.6/dist-packages/torch/tensor.py", line 195, in backward
    torch.autograd.backward(self, gradient, retain_graph, create_graph)
  File "/usr/local/lib/python3.6/dist-packages/torch/autograd/__init__.py", line 99, in backward
    allow_unreachable=True)  # allow_unreachable flag
RuntimeError: cuDNN error: CUDNN_STATUS_NOT_SUPPORTED. This error may appear if you passed in a non-contiguous input.
```

**Comment by ZwwWayne** [comment]
1. You can add the code `cfg.data.samples_per_gpu = 1` after uploading the config to modify batch size.
2. The bug may not comes from batch size. It may come from the non-local module because Pytorch 1.4 has known issues for `non-contiguous input` when using NonLocal module.
We recommend you to use Pytorch 1.5 and we already provide a [colab tutorial](https://github.com/open-mmlab/mmdetection/blob/master/demo/MMDet_Tutorial.ipynb) that helps the users to know how to install and use mmdetection including training model and modifying configs.

**Comment by CathayPacific358** [comment]
> 1. You can add the code `cfg.data.samples_per_gpu = 1` after uploading the config to modify batch size.
> 2. The bug may not comes from batch size. It may come from the non-local module because Pytorch 1.4 has known issues for `non-contiguous input` when using NonLocal module.
>    We recommend you to use Pytorch 1.5 and we already provide a [colab tutorial](https://github.com/open-mmlab/mmdetection/blob/master/demo/MMDet_Tutorial.ipynb) that helps the users to know how to install and use mmdetection including training model and modifying configs.

I dont think this tutorial is working well for now, as it keep warning that mmdet 2.1.0+e7537a4 requires torch 1.4. Then, torch 1.5.1 seems to support CUDA 10.2 while torchvision 0.6.1 supports CUDA 10.1.

**Comment by ZwwWayne** [comment]
1. I just ran the colab. The colab manually install torch1.5.1 compiled on CUDA10.1 as well as torchvision so the version should not be a problem.
2. As I said, Pytorch 1.4 has a known issue with the non-local module used by Libra R-CNN. This bug cannot be fixed in MMDet and has been fixed in Pytorch 1.5.
3. The warning should not be a problem if the model could successfully run on torch 1.5.
